ibis/backends/tests/test_client.py

```diff
@@ -202,3 +202,97 @@ def test_separate_database(con, alternate_current_database, current_data_db):
     db = con.database(current_data_db)
     assert db.name == current_data_db
     assert tmp_db.name == alternate_current_database
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_from_dataframe(con):
+
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    df = pd.DataFrame(
+        {
+            'first_name': ['A', 'B', 'C'],
+            'last_name': ['D', 'E', 'F'],
+            'department_name': ['AA', 'BB', 'CC'],
+            'salary': [100.0, 200.0, 300.0],
+        }
+    )
+
+    temp_table = 'temp_to_table'
+    con.drop_table(temp_table, force=True)
+    con.create_table(temp_table, schema=sch)
+    con.load_data(temp_table, df, if_exists='replace')
+    temporary = con.table(temp_table)
+    assert len(temporary.execute()) == 3
+
+    records = pd.DataFrame(
+        {
+            'first_name': ['X', 'Y', 'Z'],
+            'last_name': ['A', 'B', 'C'],
+            'department_name': ['XX', 'YY', 'ZZ'],
+            'salary': [400.0, 500.0, 600.0],
+        }
+    )
+
+    con.insert('temp_to_table', data_obj=records)
+    assert len(temporary.execute()) == 6
```

Comment:
I think in your first implementation you compared the dataframes, not only the number of rows in the table. That would be better if possible, so we see that INSERT is inserting the right data in the right columns.

Reply:
Right, that's better. But that requires no data to be present in the `to_table` and only then comparison can be done. Will update the code that way.